js/content/community.js

```diff
@@ -981,36 +973,34 @@ let ProfileHomePageClass = (function(){
-        let friendSteamId = rgProfileData.steamid;
+        let friendProfileData = HTMLParser.getVariableFromDom("g_rgProfileData", "object");
+        let friendSteamId = friendProfileData.steamid;
 
         HTML.replace(sendButton,
             `<span class="btn_profile_action btn_medium" id="profile_chat_dropdown_link">
-                <span>${sendButton.textContent}<img src="https://steamcommunity-a.akamaihd.net/public/images/profile/profile_action_dropdown.png"></span>
+                <span>${sendButton.textContent}<img src="//steamcommunity-a.akamaihd.net/public/images/profile/profile_action_dropdown.png"></span>
             </span>
-            <div class="popup_block" id="profile_chat_dropdown" style="visibility: visible; top: 168px; left: 679px; display: none; opacity: 1;">
-                <div class="popup_body popup_menu shadow_content" style="box-shadow: 0 0 12px #000">
-                    <a id="btnWebChat" class="popup_menu_item webchat">
-                        <img src="https://steamcommunity-a.akamaihd.net/public/images/skin_1/icon_btn_comment.png">
-                        &nbsp; ${Localization.str.web_browser_chat}
+            <div class="popup_block" id="profile_chat_dropdown">
+                <div class="popup_body popup_menu shadow_content">
+                    <a id="btnWebChat" class="popup_menu_item">
+                        <img src="//steamcommunity-a.akamaihd.net/public/images/skin_1/icon_btn_comment.png">&nbsp; ${Localization.str.web_browser_chat}
                     </a>
                     <a class="popup_menu_item" href="steam://friends/message/${friendSteamId}">
-                        <img src="https://steamcommunity-a.akamaihd.net/public/images/skin_1/icon_btn_comment.png">
-                        &nbsp; ${Localization.str.steam_client_chat}
+                        <img src="//steamcommunity-a.akamaihd.net/public/images/skin_1/icon_btn_comment.png">&nbsp; ${Localization.str.steam_client_chat}
                     </a>
                 </div>
             </div>`);
 
         document.querySelector("#btnWebChat").addEventListener("click", () => {
-            ExtensionLayer.runInPageContext(chatId => { OpenFriendChatInWebChat(chatId); }, [ chatId ]);
+            ExtensionLayer.runInPageContext(steamId => { OpenFriendChatInWebChat(steamId); }, [ friendSteamId ]);
         });
 
         document.querySelector("#profile_chat_dropdown_link").addEventListener("click", () => {
-            ExtensionLayer.runInPageContext(() => { ShowMenu(document.querySelector("#profile_chat_dropdown_link"), "profile_chat_dropdown", "right"); });
+            ExtensionLayer.runInPageContext(() => { ShowMenu("profile_chat_dropdown_link", "profile_chat_dropdown", "right"); });
+        });
+
+        document.querySelector("#profile_chat_dropdown").addEventListener("click", () => {
+            ExtensionLayer.runInPageContext(() => { HideMenu("profile_chat_dropdown_link", "profile_chat_dropdown"); });
```

Comment:
Hide the menu after clicking on an option. (Web chat opens a new window, client chat opens the client)